### ISSUE TITLE
When opening vscode Maven Projects take about 11 seconds to appear

### ISSUE THREAD

**Issue by joshunger** [Microsoft/vscode-maven#77]
**Describe the bug**
When opening vscode Maven Projects take about 11 seconds to appear.

**To Reproduce**
* Root directory has 16 directories.
* 8 subdirectories have `pom.xml` files.

1. Restart vscode.
2. Expand Maven Projects.
3. Start timer until projects appear.

**Expected behavior**
Would a second or two be acceptable?

**Environments (please complete the following information as much as possible):**
 - OS: OS X
 - VS Code version: Version 1.26.0-insider (1.26.0-insider)
 - Extension version vscjava.vscode-maven-0.9.2

**Screenshots**

**Additional context**

**Comment by Eskibear** [comment]
The extension recursively searches for `pom.xml` except for folders with certain patterns specified in `maven.projects.excludedFolders`. Do you have folders with large amount of files in your projects? If yes, you can exclude them.

There's also a setting entry `maven.projects.maxDepthOfPom`, by default it's `-1` meaning infinite, you may use that to speed up. E.g. for pom file located in `rootDir/subDir1/pom.xml`, the depth should be set at least to 3.

BTW, does the issue only occurs after you upgrade it to v0.9.2?

**Comment by joshunger** [comment]
> BTW, does the issue only occurs after you upgrade it to v0.9.2?

I don't know.

If I got these commands correct, all files -
```
➜  find . -type f | wc -l
  257431
```

Trying to exclude `node_modules` -
```
➜ find . -type f -not -path "*node_modules*" | wc -l 
   33991
```

**Comment by Eskibear** [comment]
In your case I suggest that you exclude more files by adding items into the `maven.projects.excludedFolders` for the moment. Or change the value of `maven.projects.maxDepthOfPom` to a positive integer.
```json
  // Specifies filepath pattern of folders to exclude while searching for maven projects.
  "maven.projects.excludedFolders": [
    "**/.*",
    "**/node_modules",
    "**/target",
    "**/bin"
  ],
```

I'll evaluate and investigate the performance.

**Comment by Eskibear** [comment]
@joshunger I'm refactoring part of the code, I find `vscode.workspace.findFiles` API should be much faster then simple traversal. And thus `maven.projects.maxDepthOfPom` would be no longer needed. FYI

**Comment by joshunger** [comment]
@Eskibear Awesome, thank you!

**Comment by Eskibear** [comment]
Fixed in commit https://github.com/Microsoft/vscode-maven/pull/83/commits/a7e54a8a61ab828f641d57462f11fa03060a6289

I've test in my working directory with a bunch of projects. Using VS Code's API, it takes 1600-1800ms to search `pom.xml` files now. (Showing nodes in the explore actually takes extra time, about 1 seconds in my experiment.)

```bash
$ find . -type f -not -path "*node_modules*" | wc -l
23170
```
It will be shipped in v1.0.0, you can checkout the latest code if you want to try it now.

**Comment by Eskibear** [comment]
released in v0.10.0
